### ISSUE TITLE
3.7.1 and 3.6.1 both are quitting repeatedly within about 5 seconds in OSX 10.15.7

### ISSUE THREAD

**Issue by jtieber** [ViennaRSS/vienna-rss#1447]
Vienna version 3.7.1 :d356d926: (7437), running on OSX 10.15.7, became completely non-functional late last night (MAR 4), always quitting within a few seconds of starting.

I:
• ran DiskUtility (it found nothing wrong)
• ran Malwarebytes (it found nothing wrong)
• deleted (using AppCleaner) a software program (Garvin Express [vehicle GPS]) I had installed a few hours prior to Vienna becoming non-functional
• restarted, closed all other programs, including my VPN
• started Vienna

No change and so, as I noticed the top pinned post at forums.cocoaforge (I registered about 4 hours ago and it's not yet been approved, hence I'm here),  'Vienna 3.6.1 is our current recommendation', I installed 3.6.1, then started 3.6.1, again with no other applications open and with my VPN off — 

No change.

I then re-installed 3.7.1 (downloaded from vienna-rss.com).

No change.

I've attached the error message as a PDF, in the event it's any help.

[210305 VIENNA error message.pdf](https://github.com/ViennaRSS/vienna-rss/files/6091756/210305.VIENNA.error.message.pdf)

**Comment by TAKeanice** [comment]
Looks like Vienna is refreshing and there is an incredibly deep stacktrace of XML elements being released: see the part starting at `Thread 4 Crashed:: Dispatch queue: uk.co.opencommunity.vienna2.refresh` in the log.

Another first step for debugging the problem further: Could you rename the folder `~/Library/ApplicationSupport/Vienna` and see if it starts? Vienna will create a new Vienna folder with some default feeds. You can delete the new Vienna folder later and name the old folder back to Vienna after the test.

**Comment by jtieber** [comment]
THANK YOU!

That worked.

My reading of your email is that you’ve given me just the first step. But in the event I’ve misunderstood:

I waited just a few minutes, deleted the newly created Vienna folder, renamed the renamed folder back to Vienna, and it crashed again upon opening.

I’m now back to Vienna running without crashing with the default feeds.

What should I do next to be able to revert back to my (300+ RSS, mostly non-legacy news) feeds.

John

**Comment by jtieber** [comment]
I've tried a few things, and all have failed.

How can I bring back my 301 RSS feeds?

**Comment by TAKeanice** [comment]
I would have to look into why Vienna crashes with your database. It might be due to the high number of feeds. It could also be due to a specific feed that is malformed in a way we didn’t predict. 
I am quite certain that the refresh happening at the application start causes the error. Switching that refreshing off as a next step could help narrow down the issue. For that, start Vienna like you did before so it doesn’t crash, go to settings and switch the refresh interval to manual and refresh at start to “off”. Then you can again try starting with your 300+ feeds. If it doesn’t crash without the refresh, you could try refreshing the feeds in chunks manually. If any chunk causes the crash, cut it in half, refresh both halves individually, and keep doing so until you singled out the feed that causes the crash. Then report here which feed is the culprit. If Vienna crashes for none of the chunks, try refreshing all feeds at once and report here if it crashes.

**Comment by jtieber** [comment]
Tassilo:

Thank you again.

My feeds — and the daily time I spend staring into my MacBook Air — have increased as the non-legacy news has, and so I knew some pruning needed to be done anyway, but kept putting it off (I’m getting about 6000 posts per month, and only reading/skimming about 1500). (I think I blew an appointment with a Harvard recruiter in my junior year in high school in 1969 because all I could come up with regarding hobbies was ‘reading.')

 I’ll continue working on this — and REPORT here what I did just prior to Vienna becoming fully-functional, assuming that happens and a particular action just prior can be determined.

Last night, I did, in fact, do similarly to what you suggest below. I tried moving just 10 feeds at a time into the new folder but, perhaps due to my using the wrong method, these didn’t appear when I started Vienna with the default folder. I then opened some in LIBRARY but concluded it would be very time-consuming to determine which to delete by looking at each individually by that method. So I then repeatedly deleted some of my feeds in the few seconds that was available each time Vienna started prior to it crashing. Only after completing that did I realize (duhhh 😀 ) that I could have done that much easier by shutting off 'refresh on start'.

I also deleted all my IMAGES in library (about 11 MB worth, if I recall correctly), and reindexed the database (though I don’t know if that’s relevant, and there was no indication anything was done).

I gradually reduced ‘Concurrent Downloads’ to 1, but that seemed to have no effect other than considerably slowing the refreshes, so I’m back at 50 with no change from 1.

It’s very functional at the moment; I have it set not to refresh on start; when I refresh manually, it appears to refresh everything, then immediately crashes. Then I can reopen and go through my feeds, and it USUALLY doesn’t crash until I next refresh.

John

**Comment by jtieber** [comment]
Tassilo:

Brief correction from my previous message:

I put ‘USUALLY’ in all caps below as I thought it might provide a clue.

> 

> ...Then I can reopen and go through my feeds, and it USUALLY doesn’t crash until I next refresh.

But I had forgotten that after switching to manual refresh I then switched to every 30 minutes — because, being so accustomed to automatic refreshes, I likely would have thought I was caught up on my feeds, forgetting that they were not automatically refreshing.

I just timed this: Vienna is only crashing when it refreshes.

John

**Comment by TAKeanice** [comment]
Hello again! 

It's becoming difficult to debug the problem from distance without knowing your database. If you don't have anything overly private in there, you can send it to me, so I can reproduce the behavior.

Otherwise, we should try the refreshing of a selection of feeds again. I think there was some misunderstanding. Refreshing only a bunch of feeds works by marking several ones in the sidebar, right clicking and choosing "refresh selected subscription" (wording might differ, as I am using Vienna in German)

Have a nice Sunday!

**Comment by jtieber** [comment]
> On Mar 6, 2021, at 2:26 AM, Tassilo Karge <notifications@github.com> wrote:
> 
> ... If it doesn’t crash without the refresh, you could try refreshing the feeds in chunks manually. If any chunk causes the crash, cut it in half, refresh both halves individually, and keep doing so until you singled out the feed that causes the crash. Then report here which feed is the culprit….


Tassilo:

Thanks. That consumed less than 10 minutes of my time.

It was this feed: https://home.solari.com/feed/ .

(I suspect not additionally useful, but in the event I’m wrong: screenshot of Get Info also attached.)

**Comment by jtieber** [comment]
(above sent via email - attachment didn't transmit)

![Screen Shot 2021-03-08 at 4 42 50 AM](https://user-images.githubusercontent.com/80116982/110324536-f7a77400-7fca-11eb-82a1-65b10563cd41.png)

**Comment by TAKeanice** [comment]
Could you help us to investigate further @jtieber? See #1448

**Comment by jtieber** [comment]
> On Mar 9, 2021, at 12:17 PM, Tassilo Karge ***@***.***> wrote:

> Could you help us to investigate further @jtieber <https://github.com/jtieber>? See #1448 <https://github.com/ViennaRSS/vienna-rss/issues/1448>

I’d be happy to help, but don’t know what else I can add.

I had 301 feeds; Vienna suddenly began crashing a few seconds after starting up. Initially I didn’t associate the crashing with refreshing (I’ve always had it set to refresh on starting). Tassilo suggested it was crashing only when refreshing, which turned out to be correct and, using Tassilo’s suggestion, I refreshed chunks of feeds until I identified and removed the Solari feed.

There’s been no problems since I deleted that feed.
